pkg/planner/core/point_get_plan.go

```diff
@@ -1157,6 +1160,66 @@ func tryPointGetPlan(ctx sessionctx.Context, selStmt *ast.SelectStmt, check bool
 	return nil
 }
 
+func checkGlobalIndexForPointGetPlan(ctx sessionctx.Context, tblName *ast.TableName, tblAlias model.CIStr, fields []*ast.SelectField, pairs []nameValuePair, check bool) *PointGetPlan {
```

Comment:
Why not just use code in `tryPointGetPlan`? Many lines are same as `tryPointGetPlan` function.

Maybe we only need L1167-L1181.

Reply:
How about this one? @L-maple